spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/Engine.scala

```diff
@@ -400,24 +409,46 @@ object Engine {
       throw new IllegalArgumentException(s"Engine.init: Unsupported master format $master")
     }
   }
+}
 
-  private def checkSysEnv() : Unit = {
-    // bigdl.disableCheckSysEnv is only for test purpose
-    if (System.getProperty("bigdl.disableCheckSysEnv") != null) {
-      return
+object Env {
+  val waitTimeMilliSec: Int = {
+    System.getProperty("bigdl.mklBlockTime", "0").toLowerCase(Locale.ROOT) match {
+      case x if x.toInt >= 0 => x.toInt
+      case option =>
+        throw new IllegalArgumentException(s"Unknown option of $option for bigdl.blockTime")
     }
+  }
 
-    readConf
-      .filter(_._1.startsWith("spark.executorEnv."))
-      .map(c => (c._1.substring(18), c._2))
-      .foreach(env => {
-        require(System.getenv(env._1) != null,
-          s"Engine.init: Cannot find ${env._1} in environment variables. $ENV_VAR_ERROR")
-        require(System.getenv(env._1) == env._2,
-          s"Engine.init: Environment variable ${env._1} is ${System.getenv(env._1)}. " +
-          s"But it should be ${env._2}. $ENV_VAR_ERROR")
-    })
+  val mklNumThreads: Int = {
+    System.getProperty("bigdl.mklNumThreads", "1").toLowerCase(Locale.ROOT) match {
+      case x if x.toInt > 0 => x.toInt
+      case option =>
+        throw new IllegalArgumentException(s"Unknown option of $option for bigdl.mklNumThreads")
+    }
+  }
+
+  def mklDisableFastMM(): Unit = {
+    System.getProperty("bigdl.mklDisableFastMM", "true").toLowerCase(Locale.ROOT) match {
+      case "true" => MKL.disableFastMM()
+      case "fasle" =>
+      case option =>
+        throw new IllegalArgumentException(s"Unknown option of $option for bigdl.mklDisableFastMM")
+    }
   }
 
-  checkSysEnv()
+  def waitPolicyPassive(): Unit = {
+    System.getProperty("bigdl.mklWaitPolicy", "passive").toLowerCase(Locale.ROOT) match {
+      case "passive" => MKL.waitPolicyPassive()
+      case option =>
+        throw new IllegalArgumentException(s"Unknown option of $option for bigdl.waitPolicy")
+    }
+  }
+
+  def setMKLEnv(): Unit = {
```

Comment:
You call setMKLEnv in Engine.init, which will not be called in the executors. So all executor won't set the MKL env. This is wrong.

Remove the setMKLEnv(). Call this piece of code When the class is loaded to JVM.